pkg/controller/vsphere/reconciler.go

```diff
@@ -75,6 +83,30 @@ func (r *Reconciler) create() error {
 		return fmt.Errorf("%v: failed validating machine provider spec: %w", r.machine.GetName(), err)
 	}
 
+	techPreview, err := featuregates.IsFeatureSetApplied(r.Context, v1.TechPreviewNoUpgrade, r.apiReader)
+	if err != nil {
+		return fmt.Errorf("%v: unable to check featuresets: %w", r.machine.GetName(), err)
+	}
```

Comment:
You shouldn't be gating on the `FeatureSet`, but rather a `FeatureGate`. Please add a new FeatureGate to OpenShift API and then gate on whether that is enabled or not, rather than gating on if the cluster is TechPreview. Doing this based on `FeatureSet` makes it very hard to evolve the feature later.

[FeatureGate FAQs](https://docs.google.com/document/d/1zOL38_KDKwAvsx-LMHfyDdX4-jq3HQU1YWBjfuHaM0Q/edit#) for reference